crates/bevy_text/src/lib.rs

```diff
@@ -38,6 +38,25 @@ pub type DefaultTextPipeline = TextPipeline<Entity>;
 #[derive(Default)]
 pub struct TextPlugin;
 
+/// `TextPlugin` settings
+#[derive(Resource)]
+pub struct TextSettings {
+    /// Maximum number of font atlases supported in a FontAtlasSet
+    pub max_font_atlases: usize,
+    /// Allows font size to be set dynamically exceeding the amount set in max_font_atlases.
+    /// Note each font size has to be generated which can have a strong performance impact.
+    pub allow_dynamic_font_size: bool,
```

Comment:
So on this point I disagree that we should merge this into a single field, the reason for that is that in this implementation we do not and should not support a  "there is no limit case". 
The two use cases being supported here are:
1.  Panicking if the user surpasses the max_font_atlases limit (allow_dynamic_font_size == false | default behavior)
2. Use a rotating buffer of the size max_font_atlases if the limit is exceeded (allow_dynamic_font_size == true | this behavior must be set by the user since it has a high performance cost associated with it)

In each case both fields are necessary to determine the behavior and the size of the buffer. I do agree that max_font_atlases should be changed to NonZeroUsize since there is no use case where we would want it to be 0. I have made a change to reflect that. 

 We should not support a "there is no limit" because ultimately there would leave no protection for the user from the memory leak and with these two settings they should be able to reach any desired behavior while still providing safeguards.

Reply:
Ah, I see! Okay, please feel free to resolve this comment then.